lib/internal/http2/util.js

```diff
@@ -349,10 +350,19 @@ function updateSettingsBuffer(settings) {
     settingsBuffer[IDX_SETTINGS_MAX_FRAME_SIZE] =
       settings.maxFrameSize;
   }
-  if (typeof settings.maxHeaderListSize === 'number') {
+  if (typeof settings.maxHeaderListSize === 'number' ||
+      typeof settings.maxHeaderSize === 'number') {
     flags |= (1 << IDX_SETTINGS_MAX_HEADER_LIST_SIZE);
+    if (settings.maxHeaderListSize !== undefined &&
+      settings.maxHeaderSize !== undefined &&
+      (settings.maxHeaderListSize !== settings.maxHeaderSize)) {
+      throw new ERR_HTTP2_INVALID_SETTING_VALUE.RangeError(
+        'maxHeaderListSize', settings.maxHeaderListSize
+      );
+    }
```

Comment:
```js
    if (settings.maxHeaderListSize !== undefined &&
      (settings.maxHeaderListSize !== settings.maxHeaderSize)) {
      process.emitWarning(
        'setting.maxHeaderListSize overwrite setting.maxHeaderSize'
      );
      settingsBuffer[IDX_SETTINGS_MAX_HEADER_LIST_SIZE] =
          settings.maxHeaderListSize;
    } else {
      settingsBuffer[IDX_SETTINGS_MAX_HEADER_LIST_SIZE] =
          settings.maxHeaderSize;
    }
```

Reply:
as I said https://github.com/nodejs/node/pull/33636#discussion_r432620113

Reply:
thinking twice, I prefer `maxHeaderListSize` to overwrite that. 

Imaging you use the old `http2` package if it will auto add `maxHeaderListSize` if not present, but u want to use the `maxHeaderListSize`, it's sad that if `maxHeaderListSize` will be overwritten or throw an error.

Reply:
@Himself65 I think you misspelled `maxHeaderSize` as `maxHeaderListSize` somewhere in your comment.